src/compas/geometry/offset/offset.py

```diff
@@ -185,33 +168,76 @@ def offset_polyline(polyline, distance, normal=[0.0, 0.0, 1.0]):
 
     """
 
-    p = len(polyline)
+    distances = match_value_sequence(distance, polyline)
+    segments = offset_segments(polyline, distances, normal)
+
+    offset = [segments[0][0]]
+    for s1, s2 in pairwise(segments):
+        point = intersect(s1, s2, tol)
+        offset.append(point)
+    offset.append(segments[-1][1])
+
+    return offset
+
 
-    if isinstance(distance, (list, tuple)):
-        distances = distance
+def match_value_sequence(value, sequence):
```

Comment:
not sure about the name of this thing, otherwise it looks great.
perhaps this should also become generally available like `pairwise`, `flatten`, ...

Reply:
How do `match(value, sequence)` or `match_sequence(value, sequence)` in `utilities.itertools_.py` look to you?

Reply:
not sure but perhaps it should be thebtopic of a separate PR